src/code_803E724.c

```diff
@@ -0,0 +1,1461 @@
+            u32 unkVar;
+            if (info->eyesightStatus.eyesightStatus == STATUS_BLINKER) {
+                strPtr->blinded = 1;
+                unkVar = 0xE;
+            }
+            else {
+                strPtr->blinded = unk18214;
+                unkVar = 0;
+            }
+
+            if (gUnknown_202EDFC != unkVar) {
+                gUnknown_202EDFC = unkVar;
+                SetBGOBJEnableFlags(unkVar);
+            }
+        }
+        else {
+            gUnknown_202EDFC = 0xFFFF;
+        }
+
+        if (before != strPtr->blinded) {
+            sub_8041888(0);
+        }
+
+        before = strPtr->hallucinating;
+        if (info->eyesightStatus.eyesightStatus == STATUS_CROSS_EYED) {
+            strPtr->hallucinating = 1;
+        }
+        else {
+            strPtr->hallucinating = 0;
+        }
+
+        if (before != strPtr->hallucinating) {
+            sub_806CC70();
+            sub_8041888(0);
+        }
+
+        before = strPtr->unk1820F;
+        if (CanSeeInvisibleMonsters(cameraTarget)) {
+            strPtr->unk1820F = 1;
+        }
+        else {
+            strPtr->unk1820F = 0;
+        }
+
+        if (before != strPtr->unk1820F) {
+            sub_8049ED4();
+            sub_8040A84();
+            sub_8041888(0);
+        }
+
+        if (a0 && strPtr->unk18213 == 0) {
+            if (abs(strPtr->cameraPixelPosMirror.x - strPtr->cameraPixelPos.x) > 7 || abs(strPtr->cameraPixelPosMirror.y - strPtr->cameraPixelPos.y) > 7) {
+                sub_8049ED4();
+                strPtr->unk18213 = 1;
+            }
+            else {
+                if (strPtr->cameraPixelPosMirror.x < strPtr->cameraPixelPos.x) {
+                    sub_804A1F0(240, 0);
+                }
+                else if (strPtr->cameraPixelPosMirror.x > strPtr->cameraPixelPos.x) {
+                    sub_804A1F0(0, 0);
+                }
+
+                if (strPtr->cameraPixelPosMirror.y < strPtr->cameraPixelPos.y) {
+                    sub_804A49C(0, 160);
+                }
+                else if (strPtr->cameraPixelPosMirror.y > strPtr->cameraPixelPos.y) {
+                    sub_804A49C(0, 0);
+                }
+            }
+        }
+
+        if (strPtr->cameraPos.x != strPtr->cameraPosMirror.x || strPtr->cameraPos.y != strPtr->cameraPosMirror.y) {
+            for (i = 0; i < DUNGEON_MAX_POKEMON; i++) {
+                Entity *mon = gDungeon->allPokemon[i];
+                if (EntityExists(mon)) {
+                    sub_80402AC(mon->pos.x, mon->pos.y);
+                }
+            }
+        }
+    }
+
+    sub_80400D4();
+    sub_803F7BC();
+}
+
+void sub_803F7BC(void)
+{
+    UnkDungeonGlobal_unk181E8_sub *strPtr = &gDungeon->unk181e8;
+    Tile *tile = GetTile(strPtr->cameraPos.x, strPtr->cameraPos.y);
+    u32 roomId = tile->room;
+
+    if (strPtr->unk1820B != 0 || strPtr->unk1820C != 0 || strPtr->unk18217 != 0) {
+        sub_8005838(NULL, 0);
+    }
+    else if (roomId == 0xFF) {
```

Comment:
```suggestion
    else if (roomId == CORRIDOR_ROOM) {
```
I think this is the correct define name.